hooks/hiero_update_cuts.py

```diff
@@ -0,0 +1,99 @@
+# Copyright (c) 2018 Shotgun Software Inc.
+#
+# CONFIDENTIAL AND PROPRIETARY
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+#  UI Hook
+# ===========================
+
+import sgtk
+
+HookBaseClass = sgtk.get_hook_baseclass()
+
+
+class HieroUpdateCuts(HookBaseClass):
+    """
+    This class defines methods that control if and how Cuts and CutItems
+    are created or updated during the export process.
+    """
+    def allow_cut_updates(self, preset_properties):
+        """
+        Determines whether to process the associated Cut entity during
+        export. The preset properties provided allow for customization
+        of this behavior based on custom properties added to the shot
+        processor preset via other hooks, such as the
+        customize_export_ui hook.
+
+        Example Implementation:
+
+        .. code-block:: python
+
+            # The my_custom_property is a bool property that controls
+            # whether we update cuts or not.
+            return preset_properties.get("my_custom_property", True)
+
+        :param dict preset_properties: The properties dictionary of
+            shot processor preset.
+
+        :returns: True to allow Cut updates, False to disallow.
+        :rtype: bool
+        """
+        return True
+
+    def create_cut_item(self, cut_item_data, preset_properties):
+        """
+        Handles the creation of the CutItem entity in Shotgun. This
+        hook method can be overridden in order to put conditions on
+        whether or how this creation occurs. The preset's properties
+        are provided to allow for looking up custom properties that
+        might have been added to the preset in other hooks, like can
+        be achieved when using the hiero_customize_export_ui hook.
+
+        :param dict cut_item_data: The dictionary of field/value
+            pairs to use when creating the CutItem entity in Shotgun.
+        :param dict preset_properties: The export preset's properties
+            dictionary.
+
+        :returns: The created CutItem entity dictionary, or None if
+            no CutItem entity was created.
+        :rtype: dict or None
+        """
+        cut_item = self.parent.sgtk.shotgun.create("CutItem", cut_item_data)
+        self.parent.logger.info("Created CutItem in Shotgun: %s" % cut_item)
```

Comment:
will the fact that we don't batch these guys really slow things down?
Could the hook take a list of cut items and do a batch?

Reply:
I can look at the underlying logic and see what might be possible. Is it too slow now? This isn't changing the logic flow other than running it from a hook.

Reply:
is this new or old code? I am not sure about current performance. But if you are running say a 200 shot conform, running create 200 times will definitely be slower than running a batch. But like you say, it depends on other factors, workflow ux etc. If this is old code and we are merely updating the docs, probably better to leave things the way they are!